UnitedIncome/components
# Portal -> TypeScript -> Hooks

### Description
> Please provide a brief description of your changes below.

Re-writes the `Portal` component to use a hook and TypeScript. I mostly wanted to try my hand at writing a reusable hook, it's probably not a _great_ example but we should discuss how we want to export these in the future. This can be consumed by the `Portal` component, along with the `DropDown` component when it's re-written as they share the same `createElement` logic.

### Testing Instructions
> What should be done in order to test this pull request, or re-create its effects.

* Test the changes in `client` and `dotcom` and assure they still work as intended. I will need to do some more testing myself on this on Monday.

### Comments
> Any additional comments that will aid with the review process.

Not sold on `Portal.hooks.tsx` as the filename, perhaps it should just be a named export from `Portal.tsx` so you can do `import Portal, {usePortal} from '~components/utilities/Portal/Portal`.
